regclient/search.go

```diff
@@ -49,21 +49,51 @@ func (c Client) Search(p *SearchParams) ([]*registry.Result, error) {
 	return results, nil
 }
 
-func (c Client) doJSONSearchReq(method string, s *registry.SearchParams) ([]*registry.Result, error) {
-	if c.cfg.Location == "" {
-		return nil, ErrNoRegistry
+func (c Client) prepPostReq(s *registry.SearchParams) (*http.Request, error) {
+	data, err := json.Marshal(s)
+	if err != nil {
+		return nil, err
 	}
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/search", c.cfg.Location), bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	return req, nil
+}
 
-	data, err := json.Marshal(s)
+func (c Client) prepGetReq(s *registry.SearchParams) (*http.Request, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/search", c.cfg.Location), nil)
 	if err != nil {
 		return nil, err
 	}
+	q := req.URL.Query()
+	q.Add("q", s.Q)
+	if s.Limit > 0 {
+		q.Add("limit", fmt.Sprintf("%d", s.Limit))
+	}
+	if s.Offset > -1 {
+		q.Add("offset", fmt.Sprintf("%d", s.Offset))
+	}
+	req.URL.RawQuery = q.Encode()
+	return req, nil
+}
 
-	req, err := http.NewRequest(method, fmt.Sprintf("%s/search", c.cfg.Location), bytes.NewReader(data))
+func (c Client) doJSONSearchReq(method string, s *registry.SearchParams) (results []*registry.Result, err error) {
+	if c.cfg.Location == "" {
+		return nil, ErrNoRegistry
+	}
+	req := &http.Request{}
```

Comment:
this should be: 
```go
var req *http.Request
```
This code is currently allocating an complete `http.Request` struct, allocating a pointer, and then never using the struct, as all cases in the following `switch` statement re-assigning the value of `req`.